src/fdb5/tools/fdb-list.cc

```diff
@@ -93,24 +86,20 @@ void FDBList::init(const CmdArgs& args) {
     porcelain_ = args.getBool("porcelain", false);
     json_ = args.getBool("json", false);
     compact_ = args.getBool("compact", false);
+    depth_     = args.getInt("depth", 3);
 
-    if (json_) {
-        porcelain_ = true;
-        if (location_ || timestamp_ || length_) {
-            throw UserError("--json and --location/--timestamp/--length not compatible", Here());
-        }
+    if (json_) { porcelain_ = true; }
+
+    if (porcelain_) {
```

Comment:
This is quite elegant, in that it removes the need for if (porcelain_) later on.

But I would rather that we still throw a UserError if the user asks for something else, and explain what the conflict is, rather than just give them an output that they likely don't expect.